riscv/src/code_gen.rs

```diff
@@ -319,43 +375,154 @@ fn preamble<T: FieldElement>(runtime: &Runtime, with_bootloader: bool) -> String
         + &bootloader_preamble_if_included
         + &memory(with_bootloader)
         + r#"
-    // ============== Constraint on x0 =======================
-
-    x0 = 0;
-
     // ============== iszero check for X =======================
     let XIsZero = std::utils::is_zero(X);
 
+    // =============== Register memory =======================
+    std::machines::memory::Memory regs;
+    instr get_reg Y -> X link ~> X = regs.mload(Y, STEP);
+    instr set_reg X, Y -> link ~> regs.mstore(X, STEP, Y);
+    reg val1;
+    reg val2;
+    col witness val1_col;
+    col witness val2_col;
+    col witness val3_col;
+    col witness val4_col;
+
+    col witness XX, XX_inv, XXIsZero;
+    std::utils::force_bool(XXIsZero);
+    XXIsZero * XX = 0;
```

Comment:
could use a std tool function here.

Reply:
we actually can't because in instructions that don't use these columns, everything would be set to 0 by witgen and the constraints can't be satisfied. I'll add a comment saying this